### ISSUE TITLE
High number of crashes caused by ConcurrentModificationException

### ISSUE THREAD

**Issue by angelix** [Countly/countly-sdk-android#410]
After updating the countly android sdk we see a significant increase of crashed in google play store.

### Current Behavior
```
Exception java.util.ConcurrentModificationException:
  at java.util.HashMap$HashIterator.nextNode (HashMap.java:1603)
  at java.util.HashMap$EntryIterator.next (HashMap.java:1636)
  at java.util.HashMap$EntryIterator.next (HashMap.java:1634)
  at ly.count.android.sdk.ModuleViews.startStoppedViews (ModuleViews.java:489)
  at ly.count.android.sdk.ModuleViews.onActivityStarted (ModuleViews.java:557)
  at ly.count.android.sdk.Countly.onStartInternal (Countly.java:946)
  at ly.count.android.sdk.Countly$6.onActivityStarted (Countly.java:737)
  at android.app.Application.dispatchActivityStarted (Application.java:416)
  at android.app.Activity.dispatchActivityStarted (Activity.java:1506)
  at android.app.Activity.onStart (Activity.java:2040)
  at androidx.fragment.app.FragmentActivity.onStart (FragmentActivity.java:344)
  at androidx.appcompat.app.AppCompatActivity.onStart (AppCompatActivity.java:251)
  at com.**.**.ui.MainActivity.onStart (MainActivity.kt:367)
```

### Your Environment

* Countly 24.7.4
* Android SDK 35
* Google Pixel
* Minor changes you can check here https://github.com/angelix/countly-sdk-android/tree/blockstream-green

**Comment by arifBurakDemiray** [comment]
Hello @angelix,

I attempted to reproduce this issue by launching a main activity on a similar setup (Android SDK 35 on a Google Pixel), but no errors occurred during those tests. This might suggest the error arises under specific conditions, such as particular background states, activity transitions, or configurations not present in our testing environment.

Next Steps:
To pinpoint the cause more effectively, could you let us know if there are any specific sequences of actions or user behaviors that seem to trigger this crash?

We've prepared a solution to address the ConcurrentModificationException issue by replacing the regular HashMap with a ConcurrentHashMap in the ModuleViews.startStoppedViews method. This update is designed to improve stability and prevent the error during concurrent access.

The proposed fix is available in [Pull Request #411](https://github.com/Countly/countly-sdk-android/pull/411), and we'll keep you updated on its publication. Let us know if you have any questions in the meantime!

Thank you for your help in troubleshooting this!

**Comment by arifBurakDemiray** [comment]
Hi @angelix , just checking in to see if you have any updates on the issue you reported—let us know if we can help! I am closing this issue but do not hesitate to open a new issue if issue persist. We are always happy to help. Have a good day!

**Comment by angelix** [comment]
Hi @arifBurakDemiray,

Sorry for not replying earlier.
 Based on production data the fix in #411 solved the issue.

Thanks

**Comment by arifBurakDemiray** [comment]
Hello @angelix, I am happy to hear that 😊. We are always happy to help. Have a good day!
